hax/hax/handler.py

```diff
@@ -166,72 +172,48 @@ def update_process_failure(self, planner: WorkPlanner,
-                        elif current_status == ObjHealth.OK:
-                            proc_status = (
-                                m0HaProcessEvent.M0_CONF_HA_PROCESS_STARTED)
-                    if proc_status is not None:
-                        self.consul.update_process_status(
-                            ConfHaProcess(chp_event=proc_status,
-                                          chp_type=proc_type,
-                                          chp_pid=0,
-                                          fid=state.fid))
-                    # RC or not RC, i.e. even without persistent state update,
-                    # it is important that the notification to local motr
-                    # processes must still be sent.
-                    new_ha_states.append(
-                        HAState(fid=state.fid, status=current_status))
-                if not self.consul.is_proc_local(state.fid):
-                    proc_status_local = self.consul.get_process_status_local(
-                                            state.fid)
-                    # Consul monitors a process every 1 second and this
-                    # notification is sent to every node. Thus to avoid
-                    # notifying about a process multiple times about the
-                    # same status every node maintains a local copy of the
-                    # remote process status, which is checked everytime a
-                    # consul notification is received and accordingly
-                    # the status is notified locally to all the local motr
-                    # processes.
-                    if (current_status == ObjHealth.OK and
-                            (proc_status_local.proc_status !=
-                             'M0_CONF_HA_PROCESS_STARTED')):
-                        proc_status = (
-                            m0HaProcessEvent.M0_CONF_HA_PROCESS_STARTED)
-                    elif current_status in (ObjHealth.OFFLINE,
-                                            ObjHealth.FAILED):
-                        if (proc_status_local.proc_status !=
-                                'M0_CONF_HA_PROCESS_STOPPED'):
-                            proc_status = (
-                                m0HaProcessEvent.M0_CONF_HA_PROCESS_STOPPED)
-                    else:
-                        continue
-                    if proc_status is not None:
-                        self.consul.update_process_status_local(
-                            ConfHaProcess(chp_event=proc_status,
-                                          chp_type=proc_type,
-                                          chp_pid=0,
-                                          fid=state.fid))
+                proc_status = proc_Health_to_status.get(current_status)
+                if proc_status is not None:
+                    if proc_status_remote.proc_status != proc_status.name:
+                        if (self.consul.am_i_rc() or
```

Comment:
Can we end up in situation where there is no RC(RC POD restart scenario where new RC is not elected yet)?

Reply:
Yes, so `am_i_rc()` first tries to get the leader node,
```
    def get_leader_node(self) -> str:
        """
        Returns the node name of RC leader.
        Note: in case when RC leader is not elected yet, the node name may be
        just a randomly generated string (see hare-node-join script for
        more details).
        """

        leader = self.kv.kv_get('leader', allow_null=True)
        if leader is None:
            raise HAConsistencyException('No leader key exists yet')

        node: bytes = leader['Value']
        if node is None:
            raise HAConsistencyException('No RC leader found')

        return node.decode('utf-8')
```
If leader node is not found, `HAConsistencyException` exception is raised and we repeat until leader node is `not None`, so `am_i_rc()` must return only if leader node is present, which guards us from `no rc leader present` situation.